core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala

```diff
@@ -50,7 +50,10 @@ private[mesos] class MesosClusterDispatcher(
   extends Logging {
 
   private val publicAddress = Option(conf.getenv("SPARK_PUBLIC_DNS")).getOrElse(args.host)
-  private val recoveryMode = conf.get("spark.mesos.deploy.recoveryMode", "NONE").toUpperCase()
+  // TODO(tnachen): Remove support for spark.mesos.deploy.recoverMode in 0.28.
+  private val recoveryMode = conf.getOption("spark.mesos.deploy.recoverMode").map { mode =>
```

Comment:
I would just remove support for this right now since we're not even documenting it. The next version will be Spark 2.0 so this is the right time to remove support for things that we don't want to maintain.